homeassistant/components/miele/sensor.py

```diff
@@ -726,3 +738,15 @@ def native_value(self) -> StateType:
     def options(self) -> list[str]:
         """Return the options list for the actual device type."""
         return sorted(set(STATE_PROGRAM_ID.get(self.device.device_type, {}).values()))
+
+
+class MieleTemperatureSensor(MieleSensor):
+    """Representation of a temperature sensor."""
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        ret_val = self.entity_description.value_fn(self.device)
+        if ret_val in DISABLED_TEMP_ENTITIES:
+            return None
+        return ret_val
```

Comment:
why don't you move this logic to `value_fn`?

Reply:
Because there are 6 temperature sensors (zone 1/2/3, core, core target, main target), it would result in a code duplication.